create-publication-from-doi/src/main/java/no/unit/nva/doi/fetch/MainHandler.java

```diff
@@ -146,7 +151,18 @@ private boolean urlIsValidDoi(URL url) {
     private CreatePublicationRequest getPublicationFromOtherUrl(URL url)
         throws URISyntaxException, MetadataNotFoundException {
         return metadataService.generateCreatePublicationRequest(url.toURI())
-            .orElseThrow(() -> new MetadataNotFoundException(NO_METADATA_FOUND_FOR + url));
+                   .map(request -> saveSourceAsLinkInAssociatedArtifacts(request, url))
+                   .orElseThrow(() -> new MetadataNotFoundException(NO_METADATA_FOUND_FOR + url));
+    }
+
+    private CreatePublicationRequest saveSourceAsLinkInAssociatedArtifacts(CreatePublicationRequest request, URL url) {
+        request.setAssociatedArtifacts(associatedArtifactsWithLinkToMetadataSource(url));
```

Comment:
er ikke dette et bedre sted å legge på request.getEntityDescription().setMetadataSource(URI.create(url.toString())); istedet for i newCreatePublicationRequest metoden

Reply:
Denne koden kjøres bare for ikke-doi metadata urler, men den du refererer til kjører uansett.